File: client/src/utils/fee-billing-unit.ts
export const FEE_BILLING_UNITS = [
  'per_hour',
  'per_month',
  'per_year',
  'per_berth_visit',
  'none',
] as const;

export type FeeBillingUnit = (typeof FEE_BILLING_UNITS)[number];

export const FEE_BILLING_UNIT_LABELS: Record<FeeBillingUnit, string> = {
  per_hour: 'Phí theo giờ',
  per_month: 'Phí theo tháng',
  per_year: 'Phí theo năm',
  per_berth_visit: 'Phí theo lượt đậu',
  none: 'Không thu phí (tàu công ty)',
};

export const FEE_BILLING_UNIT_AMOUNT_LABELS: Record<FeeBillingUnit, string> = {
  per_hour: 'Mức phí (VNĐ / giờ)',
  per_month: 'Mức phí (VNĐ / tháng)',
  per_year: 'Mức phí (VNĐ / năm)',
  per_berth_visit: 'Mức phí (VNĐ / lượt đậu)',
  none: 'Mức phí',
};

export function normalizeFeeBillingUnit(unit: string | null | undefined): FeeBillingUnit {
  if (unit && (FEE_BILLING_UNITS as readonly string[]).includes(unit)) {
    return unit as FeeBillingUnit;
  }
  return 'per_month';
}

/** Short suffix for tables/cards (after amount). */
export function feeBillingUnitSuffix(unit: string | null | undefined): string {
  const u = normalizeFeeBillingUnit(unit);
  if (u === 'none') {
    return '';
  }
  if (u === 'per_hour') {
    return '/giờ';
  }
  if (u === 'per_month') {
    return '/tháng';
  }
  if (u === 'per_berth_visit') {
    return '/lượt';
  }
  return '/năm';
}

export function formatFeeConfigDisplay(amount: number | string, unit: string | null | undefined): string {
  const u = normalizeFeeBillingUnit(unit);
  if (u === 'none') {
    return 'Không thu phí';
  }
  const n = Number(amount);
  const money = Number.isFinite(n) ? n.toLocaleString('vi-VN') : String(amount);
  return `${money} ₫${feeBillingUnitSuffix(unit)}`;
}

File: server/app/utils/fee_billing_unit.py
from typing import Optional

FEE_BILLING_UNITS = frozenset({'per_hour', 'per_month', 'per_year', 'per_berth_visit', 'none'})

_FEE_BILLING_UNIT_LABELS = {
    'per_hour': 'Phí theo giờ',
    'per_month': 'Phí theo tháng',
    'per_year': 'Phí theo năm',
    'per_berth_visit': 'Phí theo lượt đậu',
    'none': 'Không thu phí (tàu công ty)',
}


def normalize_fee_billing_unit(value: Optional[str]) -> str:
    u = (value or '').strip().lower()
    if u in FEE_BILLING_UNITS:
        return u
    return 'per_month'


def fee_billing_unit_label(unit: Optional[str]) -> str:
    normalized = normalize_fee_billing_unit(unit)
    return _FEE_BILLING_UNIT_LABELS[normalized]
